### ISSUE TITLE
Consider pulling in Rapid/Esri datasets

### ISSUE THREAD

**Issue by Cj-Malone** [Zverik/every_door#627]
Some datasets probably don't make sense, eg buildings. But addresses might be great.

There are some docs available, https://wiki.openstreetmap.org/wiki/Esri/ArcGIS_Datasets https://github.com/facebookmicrosites/Open-Mapping-At-Facebook/wiki/Esri-ArcGIS-FAQ

I think there may actually be multiple formats and protocols, esri feature server, osmxml bbox, etc. I don't know all the details but I can check it out.

**Comment by mnalis** [comment]
> Some datasets probably don't make sense, eg buildings. 

I agree, building shape is very hard and time consuming to determine from the ground. Building datasets are much better integrated from desktop tool with user verifying against aerial/satellite imagery (e.g. RapiD). 

> But addresses might be great.

Hmmmm. How exactly do you envision that would work? 

Currently user clicks on building (in `entrance` tab), and then click on housenumber text field, and type the address in.

The best I can see that could be implemented in EveryDoor is that on that building (instead of typing the address in text field) user would click on `Esri import` button/icon nearby, which would connect to the internet service, fetch the housenumber nearest to the location of that building, and asking user to confirm if that is correct housenumber for that building?

While that would be little better in best case scenario (2 clicks _vs._ 1 click + type number), it would be worse when it doesn't guess correctly (3 clicks + type number _vs._ 1 click + type number), **I'm not sure that for addresses it would be worth the effort** (especially unless the data is near-perfect, in which case automated import is probably better idea anyway).

---

Having **_StreetComplete-alike buttons to increase/decrease lastly inputted address-number_**, or even _**Vespucci-alike auto-address-prediction**_ would very likely be much **easier to implement, and would definitely be more universal**. And it would provide about the same speedup to the user entering addresses (unless in those specific areas addresses are thrown completely random instead of being in increasing/decreasing sequences)
